zerver/actions/message_flags.py

```diff
@@ -43,13 +44,19 @@ def do_mark_all_as_read(user_profile: UserProfile) -> int:
     )
     do_clear_mobile_push_notifications_for_ids([user_profile.id], all_push_message_ids)
 
-    msgs = UserMessage.objects.filter(user_profile=user_profile).extra(
-        where=[UserMessage.where_unread()],
+    msgs = (
+        UserMessage.objects.select_for_update(no_key=True)
```

Comment:
Can you add a comment here explaining a bit what this is doing, perhaps including why we need `no_key` here.